### PR TITLE
Add email login and remove phone login

### PR DESCRIPTION
**Pull request by cp-amisha-i**
https://github.com/user-attachments/assets/d800937e-965a-4ad4-813b-1ba6cea2be4d

https://github.com/user-attachments/assets/5f7cc66d-e2ce-4102-9489-7f9ac48b3c69

- [x] Add sign in with email btn and Redesign email login screen
- [x] Remove phone login
- [x]  Show login screen to phone login users
- [x] Handle email login from profile screen when delete account
- [x] Fix: even when user logs in again after delete account, remains deactivated
- [x] Fix: when tap on search bar in empty group tab, getting no result found view in group list 
- [x] Fix authentication issue with phone and google signup 

## Release Notes

- **New Features**
  - Introduced email-based login functionality, allowing users to log in using their email and password.
  - Added a "Forgot your password?" option for user assistance.
  - Added a new input view for invite codes in the group member joining process.

- **User Interface Enhancements**
  - Updated login view to include a new button for email login.
  - Added a dedicated email login view with input fields for email and password.
  - Enhanced user profile management by simplifying the flow and removing unnecessary OTP verification steps.

- **Localization**
  - Added new localization strings for email login and account management, including prompts for email and password input.
  - Updated existing strings to reflect the new email login option.

- **Bug Fixes**
  - Improved error handling for user authentication processes, particularly during account deletion and profile updates.

- **Removal of Phone Login**
  - Removed phone login functionality, including associated views and models, streamlining the authentication process.

These changes enhance user experience and expand login options within the application.

### CODE REVIEW

File: Data/Data/Repository/UserRepository.swift [hunk 1]
```diff
@@ -14,10 +14,16 @@ public class UserRepository: ObservableObject {
     @Inject private var storageManager: StorageManager
 
     public func storeUser(user: AppUser) async throws -> AppUser {
-        if let storedUser = try await store.fetchUserBy(id: user.id) {
-            return storedUser
+        if var fetchedUser = try await store.fetchUserBy(id: user.id) {
+            LogD("UserRepository: \(#function) User already exists in Firestore.")
+            if !fetchedUser.isActive {
```
Comment (cp-amisha-i) [inline Data/Data/Repository/UserRepository.swift:19]: Why we are making user active if the user is already deleted from the store?

File: Splito/UI/Home/Account/User Profile/UserProfileView.swift [hunk 2]
```diff
@@ -72,11 +72,6 @@ struct UserProfileView: View {
             ImagePickerView(cropOption: .square, sourceType: !viewModel.sourceTypeIsCamera ? .photoLibrary : .camera,
                             image: $viewModel.profileImage, isPresented: $viewModel.showImagePicker)
         }
-        .sheet(isPresented: $viewModel.showOTPView) {
```
Comment (cp-amisha-i) [inline Splito/UI/Home/Account/User Profile/UserProfileView.swift:75]: Have you tried deleting an account with an email login?

**Comment by cp-amisha-i** [comment]
Have you tested forget password and delete account delete account flow?

File: Splito/UI/Home/Account/User Profile/UserProfileView.swift (hunk 2 above)
Comment (cp-nirali-s) [reply Splito/UI/Home/Account/User Profile/UserProfileView.swift:75]: Yes

File: Data/Data/Repository/UserRepository.swift (hunk 1 above)
Comment (cp-nirali-s) [reply Data/Data/Repository/UserRepository.swift:19]: Because when we fetch deleted user then we need to make isActive true while restore the account.

**Pull request merged**